File: 0674_Longest_Continuous_Increasing_Subsequence/solution.java
package cn.huang.leetcode;

/*
    674. Longest Continuous Increasing Subsequence
    # 53
 */
public class LC_0674_LongestContinuousIncreasingSubsequence {
    /* DP
        if nums[i] > nums[i-1]
            dp[i] = dp[i-1] + 1
        else
            dp[i] = 1
     */
    public int findLengthOfLCIS(int[] nums) {
        if (nums == null || nums.length == 0) return 0;

        int maxlength = 1;
        int curr = 1;
        for (int i = 1; i < nums.length; i++) {
            if (nums[i - 1] < nums[i]) {
                curr++;
                maxlength = Math.max(curr, maxlength);
            } else {
                curr = 1;
            }
        }

        return maxlength;
    }
}

File: 0674_Longest_Continuous_Increasing_Subsequence/solution.cpp
/*
674. Longest Continuous Increasing Subsequence
https://leetcode.com/problems/longest-continuous-increasing-subsequence/
*/
#include <vector>

using namespace std;

class Solution
{
public:
  int findLengthOfLCIS(vector<int> &nums)
  {
    if (nums.empty())
      return 0;
    int cur = 1;
    int ans = 1;
    for (int i = 1; i < nums.size(); ++i)
    {
      if (nums[i] > nums[i - 1])
      {
        ++cur;
        ans = max(ans, cur);
      }
      else
      {
        cur = 1;
      }
    }
    return ans;
  }
};